src/MaxText/profiler.py

```diff
@@ -41,6 +46,32 @@ def __init__(self, config, offset_step=0):
     if config.profiler != "" and self.start_initial_profile_step >= config.steps:
       raise ValueError("Profiling requested but initial profiling step set past training final step")
 
+    # Set up the managed profiler. Only set it up on the first device.
+    self.managed_profiler = False if jax.process_index() != 0 else config.managed_profiler
```

Comment:
Is this saying that the manager should only run on the first device? Or that the profiles should only be collected on the first device? I am basically wondering if there is still a way to choose whether to profile every device or just a single one